components/writers/src/test/java/org/datacleaner/beans/writers/DeleteTableRowAnalyzerTest.java

```diff
@@ -0,0 +1,181 @@
+/**
+ * DataCleaner (community edition)
+ * Copyright (C) 2014 Neopost - Customer Information Management
+ *
+ * This copyrighted material is made available to anyone wishing to use, modify,
+ * copy, or redistribute it subject to the terms and conditions of the GNU
+ * Lesser General Public License, as published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+ * or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU Lesser General Public License
+ * for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this distribution; if not, write to:
+ * Free Software Foundation, Inc.
+ * 51 Franklin Street, Fifth Floor
+ * Boston, MA  02110-1301  USA
+ */
+package org.datacleaner.beans.writers;
+
+import static org.hamcrest.Matchers.equalTo;
+import static org.hamcrest.Matchers.is;
+import static org.junit.Assert.assertThat;
+
+import java.io.File;
+import java.util.Arrays;
+import java.util.Set;
+
+import org.apache.metamodel.UpdateCallback;
+import org.apache.metamodel.UpdateScript;
+import org.apache.metamodel.UpdateableDataContext;
+import org.apache.metamodel.data.DataSet;
+import org.apache.metamodel.schema.ColumnType;
+import org.apache.metamodel.schema.Table;
+import org.apache.metamodel.util.FileHelper;
+import org.datacleaner.api.ComponentContext;
+import org.datacleaner.api.InputColumn;
+import org.datacleaner.connection.DatastoreConnection;
+import org.datacleaner.connection.FileDatastore;
+import org.datacleaner.connection.JdbcDatastore;
+import org.datacleaner.connection.UpdateableDatastoreConnection;
+import org.datacleaner.data.MockInputColumn;
+import org.datacleaner.data.MockInputRow;
+import org.datacleaner.descriptors.AnalyzerDescriptor;
+import org.datacleaner.descriptors.Descriptors;
+import org.datacleaner.descriptors.MetricDescriptor;
+import org.easymock.EasyMock;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+public class DeleteTableRowAnalyzerTest {
+    
+    private static final String TEST_TABLE_NAME = "test_table";
+    private static final String VARCHAR_COLUMN_NAME = "foo";
+    private static final String INTEGER_COLUMN_NAME = "bar";
+    private static final String VARCHAR_COLUMN_VALUE = "StringValue";
+    private static final int INTEGER_COLUMN_VALUE = 1;
+    
+
+    private JdbcDatastore jdbcDatastore;
+    
+    public ExpectedException expectedException = ExpectedException.none();
+    
+    @Before
+    public void setUp() throws Exception {
+            jdbcDatastore = new JdbcDatastore("my datastore",
+                    "jdbc:hsqldb:mem:InsertIntoTable_testErrorHandlingOption", "org.hsqldb.jdbcDriver");
```

Comment:
This is pretty heavy for a unittest, while MetaModel allows you to use e.g. a PojoDataContext which is neat because it runs entirely in memory.

But it's not a blocker IMO.